### PR TITLE
gh-117431: Adapt str.find and friends to use the METH_FASTCALL calling convention

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-117431

### CODE REVIEW

File: Misc/NEWS.d/next/Core and Builtins/2024-04-02-17-37-35.gh-issue-117431.vDKAOn.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Improve the performance of the following :class:`str` methods
+by adapting them to the :c:macro:`METH_FASTCALL` calling convention:
+
+* :meth:`~str.endswith`
+* :meth:`~str.find`
+* :meth:`~str.index`
+* :meth:`~str.rfind`
+* :meth:`~str.rindex`
+* :meth:`~str.startswith`
```
Comment (Eclips4) [inline Misc/NEWS.d/next/Core and Builtins/2024-04-02-17-37-35.gh-issue-117431.vDKAOn.rst:10]: This PR also contains a changes related to `str.count`. Is it worth mentioning this in a `NEWS` entry?

Comment (erlend-aasland) [reply Misc/NEWS.d/next/Core and Builtins/2024-04-02-17-37-35.gh-issue-117431.vDKAOn.rst:10]: Yes, thanks for the catch!

**Comment by erlend-aasland** [comment]
Similar speedsups as with `str.startswith` and `str.endswith`:

```console
# main
$ ./python.exe -m timeit -s "s = 'abcdef'" "s.count('a')"
5000000 loops, best of 5: 92.5 nsec per loop

# This PR
$ ./python.exe -m timeit -s "s = 'abcdef'" "s.count('a')"
10000000 loops, best of 5: 26.2 nsec per loop
```

**Review by methane — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thanks for the review, Inada-san.

**Pull request merged**